File: Chapter 02/JS/2.3.4 Performance/dashboardBefore.js
const databaseService = require('./databaseService');

function getDashboardSummaryBefore(userId) {
    // Brute Force. Do the slow work every single time...
    const profile = databaseService.getProfile(userId);
    const orders = databaseService.getOrders(userId);
    const activity = databaseService.getActivity(userId);
    
    return { profile, orders, activity };
}

module.exports = getDashboardSummaryBefore;

File: Chapter 02/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter02/section_2_3_4_performance/DashboardBefore.java
package com.grokkingsoftwarearchitecture.chapter02.performance;

import java.util.HashMap;
import java.util.Map;

public class DashboardBefore {
    private final DatabaseService databaseService = new DatabaseService();

    public Map<String, String> getDashboardSummary(String userId) {
        // Brute Force. Do the slow work every single time...
        String profile = databaseService.getProfile(userId);
        String orders = databaseService.getOrders(userId);
        String activity = databaseService.getActivity(userId);

        Map<String, String> dashboardData = new HashMap<>();
        dashboardData.put("profile", profile);
        dashboardData.put("orders", orders);
        dashboardData.put("activity", activity);

        return dashboardData;
    }
}